src/mono/wasm/runtime/marshal-to-cs.ts

```diff
@@ -315,13 +315,16 @@ function _marshal_task_to_cs(arg: JSMarshalerArgument, value: Promise<any>, _?:
     }
     mono_check(isThenable(value), "Value is not a Promise");
 
-    const gc_handle = alloc_gcv_handle();
-    set_gc_handle(arg, gc_handle);
-    set_arg_type(arg, MarshalerType.Task);
+    const handleIsPreallocated = get_arg_type(arg) == MarshalerType.Task;
```

Comment:
Why is this `MarshalerType.Task` and not `MarshalerType.TaskPreCreated`?

Reply:
Good point, it will clarify things further.